src/service/viewer-impl.js

```diff
@@ -885,8 +885,12 @@ export class Viewer {
     if (eventType == 'viewport') {
       if (data['paddingTop'] !== undefined) {
         this.paddingTop_ = data['paddingTop'];
+        this.viewportObservable_.fire({
```

Comment:
To avoid obfuscation: viewer protocols are all string based, while internal protocols are we typically try to keep obfuscatable via typedefs. That being said, I don't feel strongly about this. We can push these events a bit deeper in their original form.